saw-core-coq/src/SAWCoreCoq/Term.hs

```diff
@@ -579,6 +599,35 @@ translateBinder n ty@(asPiList -> (args, pi_body)) f =
                 helper n' rest (g . ((nh,nhty) :))
          helper n' ((False,_):rest) g = helper n' rest g
 
+-- | Given an 'ExtCns' with its type (as a 'Term'), translate the 'ExtCns'
+-- to a Coq identifier, translate the type to a Coq term, and generate zero or
+-- more additional 'Ident's and 'Type's representing additonal implicit
+-- typeclass arguments, added if the given type is @isort@, etc. Pass all of
+-- this information to the supplied computation, in which the SAW core variable
+-- is bound to its Coq identifier.
+translateBinderEC ::
+  TermTranslationMonad m => ExtCns Term -> (BindTrans -> m a) -> m a
+translateBinderEC ec f =
+  do ty' <- translateTerm ty
+     let mb_sort = asSortWithFlags pi_body
+         flagValues = sortFlagsToList $ maybe noFlags snd mb_sort
```

Comment:
It might be worth leaving a comment saying that this list is guaranteed to be exactly two elements long, which is why the call to `zip flagValues flagLocalNames` below will always work as expected.

Reply:
Fixed in b4e344f69f8fb86660bdc1db30a317efef461d6e.